src/main/java/org/isf/menu/gui/UserBrowsing.java

```diff
@@ -121,250 +86,239 @@ public void userUpdated(AWTEvent e) {
+				stepPanel = new JPanel(new GridLayout(2, 1, 5, 5));
+				stepPanel.add(new JLabel(MessageBundle.getMessage("angal.userbrowser.step2.repeat.password.label")));
+				stepPanel.add(pwd);
+				int action = JOptionPane
+						.showConfirmDialog(UserBrowsing.this, stepPanel, MessageBundle.getMessage("angal.userbrowser.reset.password.title"),
+								JOptionPane.OK_CANCEL_OPTION, JOptionPane.PLAIN_MESSAGE);
+				if (JOptionPane.CANCEL_OPTION == action) {
+					return;
+				}
+				String newPassword2 = new String(pwd.getPassword());
+
+				// 3. Check & Save
+				if (!newPassword.equals(newPassword2)) {
+					MessageDialog.error(UserBrowsing.this, "angal.userbrowser.passwords.do.not.match.msg");
+					return;
+				}
+				String hashed = BCrypt.hashpw(newPassword, BCrypt.gensalt());
+				user.setPasswd(hashed);
+				try {
+					if (manager.updatePassword(user)) {
+						MessageDialog.info(UserBrowsing.this, "angal.userbrowser.password.changed.msg");
 					}
-				}	 				
+				} catch (OHServiceException e) {
+					OHServiceExceptionUtil.showMessages(e);
+				}
 			}
 		});
 		buttonPanel.add(buttonResetPassword);
-		
-		JButton buttonDelete = new JButton(MessageBundle.getMessage("angal.common.delete"));
-		buttonDelete.setMnemonic(KeyEvent.VK_D);
-		buttonDelete.addActionListener(new ActionListener() {
-			public void actionPerformed(ActionEvent event) {
-				if (table.getSelectedRow() < 0) {
-					JOptionPane.showMessageDialog(				
-	                        null,
-	                        MessageBundle.getMessage("angal.common.pleaseselectarow"),
-	                        MessageBundle.getMessage("angal.hospital"),
-	                        JOptionPane.PLAIN_MESSAGE);
-				} else {
-				User m = (User)(((UserBrowserModel) model).getValueAt(table.getSelectedRow(), -1));
-				int n = JOptionPane.showConfirmDialog(
-                        null,
-                        MessageBundle.getMessage("angal.menu.deleteuser")+" \""+m.getUserName()+"\" ?",
-                        MessageBundle.getMessage("angal.hospital"),
-                        JOptionPane.YES_NO_OPTION);
-
-                    try {
-                        if ((n == JOptionPane.YES_OPTION) && (manager.deleteUser(m))){
-                            pUser.remove(table.getSelectedRow());
-                            model.fireTableDataChanged();
-                            table.updateUI();
-                            }
-                    } catch (OHServiceException e) {
-                        OHServiceExceptionUtil.showMessages(e);
-                    }
-                }
+
+		JButton buttonDelete = new JButton(MessageBundle.getMessage("angal.common.delete.btn"));
+		buttonDelete.setMnemonic(MessageBundle.getMnemonic("angal.common.delete.btn.key"));
+		buttonDelete.addActionListener(event -> {
+			if (table.getSelectedRow() < 0) {
+				MessageDialog.error(null, "angal.common.select.row.msg");
+			} else {
+				User m = (User) model.getValueAt(table.getSelectedRow(), -1);
```

Comment:
Shall we give more meaningful name to the variable `m`?

Reply:
done

Reply:
Thanks!